factory/ideas.php

```diff
@@ -265,9 +253,29 @@ public function get_idea($id)
 	}
 
 	/**
+	 * Returns an idea specified by its topic ID.
+	 *
+	 * @param int $id The ID of the idea to return.
+	 *
+	 * @return array The idea
+	 */
+	public function get_idea_by_topic_id($id)
```

Comment:
If these were not a query searching "where topic_id = id" then sure... But this query can have multiple variations and so its not really cacheable as each cache would be overwritten by the next idea viewed. It's just 1 extra query here, and its a tiny query searching for an index so it should be ok.